nemo/collections/tts/modules/aligner.py

```diff
@@ -151,13 +157,12 @@ def forward(self, queries, keys, mask=None, attn_prior=None, conditioning=None):
             keys (torch.tensor): B x C2 x T2 tensor (text data).
             mask (torch.tensor): B x T2 x 1 tensor, binary mask for variable length entries (True = mask element, False = leave unchanged).
             attn_prior (torch.tensor): prior for attention matrix.
-            conditioning (torch.tensor): B x T2 x 1 conditioning embedding
+            conditioning (torch.tensor): B x 1 x C2 conditioning embedding
         Output:
             attn (torch.tensor): B x 1 x T1 x T2 attention mask. Final dim T2 should sum to 1.
             attn_logprob (torch.tensor): B x 1 x T1 x T2 log-prob attention mask.
         """
-        if conditioning is not None:
-            keys = keys + conditioning.transpose(1, 2)
+        keys = self.cond_input(keys.transpose(1, 2), conditioning)
```

Comment:
Yes. The output needs to be transposed back to B x C2 x T2. 
For `forward_time_minor()`, can you please explain more?

Reply:
We have inconsistent conventions where most public methods are "time major" (B X T x C) and most layer definitions are "time minor" (B x C x T).

So we have lots of inline transposes like https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/tts/modules/fastpitch.py#L91

So was wondering if it would be worth the layers having a second method like:

```
def forward_time_minor(self, input):
    input = input.transpose(1, 2)
    output = self.forward(input)
    output = output.transpose(1, 2)
    return output
```

Reply:
I see. Thanks for the suggestion!